### ISSUE TITLE
Call `parent()` for collapsed node not working

### ISSUE THREAD

**Issue by Does666** [iVis-at-Bilkent/cytoscape.js-expand-collapse#79]
I find that `parent()`  function not working when called on collapsed node.
After a quick debugging, I find it's a issue relevant to cytoscape.js versions larger than 3.3.x.
I have filed this issue [here](https://github.com/cytoscape/cytoscape.js/issues/2465). but the repo collaborator closed my issue and let me refile here.

My scenario is that I expect to search a node and highlight it,  and if its parent or even ancestor was collapsed I will expand them until the node is shown and then highlight it.

```
var node = api.getAllCollapsedChildrenRecursively().nodes('node[id="targetnode"]'), parent;
if(node.length > 0){
	parent = node;
	while(parent.length > 0){
		tmp = parent.parent();// not working on cytoscape.js over 3.4.x
		if(tmp.length > 0){
			parent = tmp;
		}else{
			break;
		}
	}
	
	api.expandRecursively(parent);
	viewUtilities.highlightNeighbors(node);
}
```

I find the key code finally that is the function removeChildRef in cytoscape.js/src/collection/index.js.
It added the code ele._private.parent = null; // remove child => parent ref in version 3.4.x while ok before.

So how do I step forward to finish my work ?

**Comment by ugurdogrusoz** [comment]
@Does666 Could you try the latest on the "unstable" branch and let us know if you observe the same behavior?

**Comment by Does666** [comment]
@ugurdogrusoz I have tried on the unstable branch, but this issue isn't resolved on it. Could you please shed light on my question ?

**Comment by hasanbalci** [comment]
@Does666 We now provided a new functionality in the api to reach parent of a node (`api.getParent(nodeId)`). It is available in the unstable branch and we'll make a release soon. 
In your code, the following line should work to get the parent node: 

`tmp = api.getParent(parent.id())`

**Comment by Does666** [comment]
@hasanbalci Looking forward to it. Thank you very much.

**Comment by ugurdogrusoz** [comment]
@Does666 Could you please try this on the latest unstable and let us know if it works? We will then make a release. Thanks

**Comment by Does666** [comment]
@ugurdogrusoz What a pity that it don't work well for me. After a quick debugging, I find that set parentData `setScratch(cy, 'parentData', parentData); ` only at  init core postion. 
My data is got from webserver when expand a node and call `cy.add()` to add to graph dynamically.   In my case, Do I have to call `api = expandCollapse({})`  every time change set data set?

**Comment by hasanbalci** [comment]
@Does666 Yes, parent information is stored when the extension is initialized. Since your graph elements are added dynamically, calling `api = expandCollapse({})` when new nodes are added can solve your problem.

**Comment by Does666** [comment]
@hasanbalci That can't solve my issue. Since `parentData` is not a global variable, every time I call `expandCollapse({})` , it will be overwriten. On the other hand, It's not a good ideal to call `expandCollapse({})` after adding new nodes which looks weird. Is it possible to encapsulate this logical into this extension ?

**Comment by hasanbalci** [comment]
@Does666 I made another fix 5385d84ba9fce716d79c23142f100c9695b0b3f3 to consider nodes added later on. Before since we keep node-parent info only for the initial graph, extension was not consider the nodes added later on. Now, it dynamically updates its content based on removed nodes. This fix may solve your issue, but if not we are open to pull requests.

**Comment by Does666** [comment]
@hasanbalci That works well for me . Thank you.
